spec/system/provider_interface/manage_provider_user_permissions_spec.rb

```diff
@@ -90,13 +95,31 @@ def when_i_remove_a_permissions_from_a_provider_user
     click_on 'Update providers'
   end
 
-  def then_i_cant_see_the_permission_for_the_provider_user
+  def then_i_cant_see_the_manage_users_permission_for_the_provider_user
     expect(page).to have_content 'Providers updated'
     expect(page).not_to have_content 'Manage users'
 
     expect(@managed_user.provider_permissions.first.manage_users).to be false
   end
 
+  def when_i_add_permission_to_view_safeguarding_for_a_provider_user
+    expect(page).not_to have_checked_field 'View safeguarding information'
+
+    within(permissions_fields_id_for_provider(@provider)) do
+      check 'View safeguarding information'
+    end
+
+    click_on 'Update providers'
+  end
+
+  def then_i_can_see_the_view_safeguarding_permission_for_the_provider_user
+    within("#provider-#{@provider.id}-enabled-permissions") do
+      expect(page).to have_content 'View safeguarding information'
+    end
+
+    expect(@managed_user.provider_permissions.first.view_safeguarding_information).to be true
```

Comment:
It would be good not to need to inspect the DB as part of a system spec. Could we assert that the checkbox is checked?

Reply:
I agree, I think this was a pattern I adopted to debug the system specs quickly but we should use expect the UI to reflect db state. Line 117 does this as it's the output from the permissions list component.